src/EditorFeatures/CSharpTest/AddRequiredParentheses/AddRequiredParenthesesTests.cs

```diff
@@ -0,0 +1,434 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.AddRequiredParentheses;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.AddRequiredParentheses;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.AddRequiredParentheses
+{
+    public partial class AddRequiredParenthesesTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (new CSharpAddRequiredParenthesesDiagnosticAnalyzer(), new AddRequiredParenthesesCodeFixProvider());
+        
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddRequiredParentheses)]
+        public async Task TestArithmeticPrecedence()
+        {
+            await TestInRegularAndScript1Async(
+@"class C
+{
+    void M()
+    {
+        int x = 1 + 2 $$* 3;
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        int x = 1 + (2 * 3);
+    }
+}", parameters: new TestParameters(options: RequireAllParenthesesForClarity));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddRequiredParentheses)]
+        public async Task TestNoArithmeticOnLowerPrecedence()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        int x = 1 $$+ 2 * 3;
+    }
+}", parameters: new TestParameters(options: RequireAllParenthesesForClarity));
```

Comment:
Nit: given how common this pattern is, I'd consider adding an overload for `TestMissingAsync` that takes a set of options. That overload could be local to this test, or possibly pushed into the test base.

Reply:
Done.  #resolved.